forc-pkg/src/pkg.rs

```diff
@@ -271,18 +279,23 @@ impl BuildPlan {
             graph,
             path_map,
             compilation_order,
+            removed_deps: vec![],
         })
     }
 
     /// Attempt to load the build plan from the `Lock`.
     pub fn from_lock(proj_path: &Path, lock: &Lock, sway_git_tag: &str) -> Result<Self> {
-        let graph = lock.to_graph()?;
+        let mut graph = lock.to_graph()?;
+        let tmp_compilation_order = compilation_order(&graph)?;
+        let (path_map, removed_deps) =
+            graph_to_path_map(proj_path, &mut graph, &tmp_compilation_order, sway_git_tag)?;
```

Comment:
It would be nice if we could keep `graph_to_path_map` as a pure function, particularly as its name doesn't really imply any modification of the `graph` itself.

Perhaps rather than mutating the `graph` *during* this function, `graph_to_path_map` can return the set of removed dependencies as something like a `HashSet<NodeIx>`, and then this can be used to update the `graph` in a following step?